### PR TITLE
Revert in progress chunked uploads when aborting them

### LINKED ISSUE #433: Revert url is only called after chunk uploads are complete, not during uploads?
Revert url is only called after chunk uploads are complete and you press the delete button. Is there a way to during chunk uploads processing when cancelled.

### PR DESCRIPTION
**Pull request by maxmitti**
I ported #497 to master as a base. As it turned out it only adds reverting for failed uploads, but not for still ongoing ones.
But that was an easy fix by just making sure the transferId is not unset before calling revert.

This fixes #433 and #755.

Note: The revert test from #497 detected well when I broke reverting of failed uploads, but does not cover still ongoing ones.
I do not have enough insight into jest and filepond to add tests for that case.

### CODE REVIEW

**Comment by rikschennink** [comment]
Thanks! 🙏 I'm on a short trip till Monday next week and will try to reserve some time then to look at this.

**Comment by maxmitti** [comment]
I might have just experienced a bug caused by this PR, but I will have to check again what exactly was happening.

I started a chunked upload and to the first patch request the server responded with 400.
This caused FilePond to send DELETE requests which also caused 400 in this instance.
After trying DELETE multiple times and maybe also more patch requests it finally stopped and showed the upload as having failed. However when trying to resume it caused an error along the lines of `Can not set property Upload-Length on undefined`.

**Comment by maxmitti** [comment]
Hmm, I tried to let the server respond with 400 deliberately, but I could not reproduce the behavior anymore.
So, maybe it was caused by some other changes I made.

**Comment by maxmitti** [comment]
@rikschennink did you have time to look at it?

**Comment by rikschennink** [comment]
@maxmitti not yet, sorry, so busy, it's in my inbox todo.

File: src/js/app/utils/createItem.js [hunk 1]
```diff
@@ -264,7 +264,6 @@ export const createItem = (origin = null, serverFileReference = null, file = nul
             state.activeProcessor = null;
 
             // if file was uploaded but processing was cancelled during perceived processor time store file reference
-            state.transferId = null;
```
Comment (rikschennink) [inline src/js/app/utils/createItem.js:267]: shouldn't this be kept?

Comment (maxmitti) [reply src/js/app/utils/createItem.js:267]: I don't actually know why the transfer id has to be cleared.
But I do know that I need to keep the id for the revert which is going to happen next.

**Pull request merged**